azkaban-webserver/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java

```diff
@@ -223,90 +223,89 @@ private void handleAJAXAction(HttpServletRequest req,
-        ajaxFetchFlowGraph(project, ret, req);
-      }
-    } else if (ajaxName.equals("fetchflownodedata")) {
-      if (handleAjaxPermission(project, user, Type.READ, ret)) {
-        ajaxFetchFlowNodeData(project, ret, req);
-      }
-    } else if (ajaxName.equals("fetchprojectflows")) {
-      if (handleAjaxPermission(project, user, Type.READ, ret)) {
-        ajaxFetchProjectFlows(project, ret, req);
-      }
-    } else if (ajaxName.equals("changeDescription")) {
-      if (handleAjaxPermission(project, user, Type.WRITE, ret)) {
-        ajaxChangeDescription(project, ret, req, user);
-      }
-    } else if (ajaxName.equals("getPermissions")) {
-      if (handleAjaxPermission(project, user, Type.READ, ret)) {
-        ajaxGetPermissions(project, ret);
-      }
-    } else if (ajaxName.equals("getGroupPermissions")) {
-      if (handleAjaxPermission(project, user, Type.READ, ret)) {
-        ajaxGetGroupPermissions(project, ret);
-      }
-    } else if (ajaxName.equals("getProxyUsers")) {
-      if (handleAjaxPermission(project, user, Type.READ, ret)) {
-        ajaxGetProxyUsers(project, ret);
-      }
-    } else if (ajaxName.equals("changePermission")) {
-      if (handleAjaxPermission(project, user, Type.ADMIN, ret)) {
-        ajaxChangePermissions(project, ret, req, user);
-      }
-    } else if (ajaxName.equals("addPermission")) {
-      if (handleAjaxPermission(project, user, Type.ADMIN, ret)) {
-        ajaxAddPermission(project, ret, req, user);
-      }
-    } else if (ajaxName.equals("addProxyUser")) {
-      if (handleAjaxPermission(project, user, Type.ADMIN, ret)) {
-        ajaxAddProxyUser(project, ret, req, user);
-      }
-    } else if (ajaxName.equals("removeProxyUser")) {
-      if (handleAjaxPermission(project, user, Type.ADMIN, ret)) {
-        ajaxRemoveProxyUser(project, ret, req, user);
-      }
-    } else if (ajaxName.equals("fetchFlowExecutions")) {
-      if (handleAjaxPermission(project, user, Type.READ, ret)) {
-        ajaxFetchFlowExecutions(project, ret, req);
-      }
-    } else if (ajaxName.equals("fetchLastSuccessfulFlowExecution")) {
-      if (handleAjaxPermission(project, user, Type.READ, ret)) {
-        ajaxFetchLastSuccessfulFlowExecution(project, ret, req);
-      }
-    } else if (ajaxName.equals("fetchJobInfo")) {
-      if (handleAjaxPermission(project, user, Type.READ, ret)) {
-        ajaxFetchJobInfo(project, ret, req);
-      }
-    } else if (ajaxName.equals("setJobOverrideProperty")) {
-      if (handleAjaxPermission(project, user, Type.WRITE, ret)) {
-        ajaxSetJobOverrideProperty(project, ret, req);
-      }
     } else {
-      ret.put("error", "Cannot execute command " + ajaxName);
+      ret.put("projectId", project.getId());
+      String ajaxName = getParam(req, "ajax");
+      if (ajaxName.equals("getProjectId")) {
+        // Do nothing, since projectId is added to all AJAX requests.
+      } else if (ajaxName.equals("fetchProjectLogs")) {
```

Comment:
Would it work using a table that maps strings to objects that implements an interface?

In any case, since this is the existing code you just moved around, your change looks fine to me.